app/models/piece.rb

```diff
@@ -124,8 +124,9 @@ def never_moved?
     updated_at == created_at
   end
 
-  # moves piece to the destination square
-  def move_to!(x,y)
+  def move_to!(params)
+    x = params[:x_position].to_i
```

Comment:
You should do this in the controller, not the model, because the model cares about x and y, and the controller cares about HTTP requests and responses (params).